src/chargepoint/reservation/ReservationManager.cpp

```diff
@@ -316,7 +324,7 @@ void ReservationManager::checkExpiries()
     // Check reservations
     for (const Connector* connector : m_connectors.getConnectors())
     {
-        if ((connector->status == ChargePointStatus::Reserved) && (connector->reservation_expiry_date <= now))
+        if ((connector->reservation_id_tag.length() > 0) && (connector->reservation_expiry_date <= now))
```

Comment:
Therre are more places where the suggestion of using `!connector->reservation_id_tag.empty()`, like here.